app/src/main/java/ca/klapstein/baudit/Data/Patient.java

```diff
@@ -1,13 +1,74 @@
 package ca.klapstein.baudit.Data;
 
+import java.util.Iterator;
+
 /**
  * Class that represents a Patient.
  */
 public class Patient extends User {
     private static final String TAG = "Patient";
-    private ProblemTreeSet problemTreeSet;
+    private ProblemTreeSet problemTreeSet = new ProblemTreeSet();
+
+    // UC-01.01.01: add problem
+    public void addProblem(Problem problem){ problemTreeSet.add(problem); }
```

Comment:
many of these methods are already given by using a `TreeSet`. 

Unless you are adding extra safety in getting/setting elements within the TreeSet I would not implement these methods.